Connecting chats

emalsidog opened the pull request.

bubuzavrik reviewed (changes requested):
About your questions:
1. We don't need roomId and subject in url, adding it was my fault. Same redirect to null was invalid realisation cause i think that subjects will be static, but then i find that we will have chat settings, where we can change folders. I've already removed that on master. 

2. You done right with the splitting header. But it would be better if you create wrapper, that will show header for us and show chat or list depending on id of conversation that stored in conversations.room.id

3. As you said currently opened room - in redux: conversations.room.

4. About stretching messages container : 
You can use:
- height in vh {
height: 80vh;
overflow: auto
}
or 
-position fixed:
{
    height: 500px;
    overflow: auto;
    position: fixed;
    bottom: 0;
}
I think that first variant will be better.
Also can you add scrollbar styling, i think it will be much prettier.

Don't forget to pull from master at first.

emalsidog commented:
> About your questions:
> 
> 1. We don't need roomId and subject in url, adding it was my fault. Same redirect to null was invalid realisation cause i think that subjects will be static, but then i find that we will have chat settings, where we can change folders. I've already removed that on master.
> 2. You done right with the splitting header. But it would be better if you create wrapper, that will show header for us and show chat or list depending on id of conversation that stored in conversations.room.id
> 3. As you said currently opened room - in redux: conversations.room.
> 4. About stretching messages container :
>    You can use:
> 
> * height in vh {
>   height: 80vh;
>   overflow: auto
>   }
>   or
> * position fixed:
>   {
>   height: 500px;
>   overflow: auto;
>   position: fixed;
>   bottom: 0;
>   }
>   I think that first variant will be better.
>   Also can you add scrollbar styling, i think it will be much prettier.
> 
> Don't forget to pull from master at first.

I have conflicts when pulling from master

bubuzavrik commented:
> > About your questions:
> > 
> > 1. We don't need roomId and subject in url, adding it was my fault. Same redirect to null was invalid realisation cause i think that subjects will be static, but then i find that we will have chat settings, where we can change folders. I've already removed that on master.
> > 2. You done right with the splitting header. But it would be better if you create wrapper, that will show header for us and show chat or list depending on id of conversation that stored in conversations.room.id
> > 3. As you said currently opened room - in redux: conversations.room.
> > 4. About stretching messages container :
> >    You can use:
> > 
> > 
> > * height in vh {
> >   height: 80vh;
> >   overflow: auto
> >   }
> >   or
> > * position fixed:
> >   {
> >   height: 500px;
> >   overflow: auto;
> >   position: fixed;
> >   bottom: 0;
> >   }
> >   I think that first variant will be better.
> >   Also can you add scrollbar styling, i think it will be much prettier.
> > 
> > Don't forget to pull from master at first.
> 
> I have conflicts when pulling from master

Conflicts resolved, try to pull it

emalsidog commented:
> > > About your questions:
> > > 
> > > 1. We don't need roomId and subject in url, adding it was my fault. Same redirect to null was invalid realisation cause i think that subjects will be static, but then i find that we will have chat settings, where we can change folders. I've already removed that on master.
> > > 2. You done right with the splitting header. But it would be better if you create wrapper, that will show header for us and show chat or list depending on id of conversation that stored in conversations.room.id
> > > 3. As you said currently opened room - in redux: conversations.room.
> > > 4. About stretching messages container :
> > >    You can use:
> > > 
> > > 
> > > * height in vh {
> > >   height: 80vh;
> > >   overflow: auto
> > >   }
> > >   or
> > > * position fixed:
> > >   {
> > >   height: 500px;
> > >   overflow: auto;
> > >   position: fixed;
> > >   bottom: 0;
> > >   }
> > >   I think that first variant will be better.
> > >   Also can you add scrollbar styling, i think it will be much prettier.
> > > 
> > > Don't forget to pull from master at first.
> > 
> > 
> > I have conflicts when pulling from master
> 
> Conflicts resolved, try to pull it

It worked :)

bubuzavrik commented on client/src/components/Messages/ConversationsWrapper/index.js:
```diff
@@ -0,0 +1,22 @@
+  return (
+    <React.Fragment>
+      <Header />
+      {Object.keys(room).length > 0 ? <Thread /> : <ConversationsList />}
```
Object.keys(room).length always more than 0, cause room it's object that have some keys, you can see all of them in reducer. So Thread will be rendered always.

emalsidog replied:
I have tried to log it in the console and it was empty. Even with default values in the reducer

bubuzavrik replied:
![image](https://user-images.githubusercontent.com/22328251/129531213-1a958e55-b744-4263-b14b-2e7040fbccf6.png)

bubuzavrik replied:
Screen with default values

bubuzavrik replied:
You need to check id, cause chats will use this id for messaging

emalsidog replied:
Ok, get it

bubuzavrik commented on client/src/components/Messages/Thread/index.js:
```diff
@@ -0,0 +1,139 @@
+
+// Styles
+import useStyles from "./styles";
+import { NoData } from "../../UI/NoData";
```
28 string it's not styles

bubuzavrik commented on client/src/components/Messages/Thread/index.js:
```diff
@@ -0,0 +1,139 @@
+    handleSettingsClose();
+  };
+
+  const settingsMenu = [
```
This array can be mapped. It will make code size lesser

bubuzavrik commented on client/src/redux/reducers/conversation.js:
```diff
@@ -7,10 +7,11 @@ import {
-import { FAILED, SUCCESS, WAITING } from '.'
-import { GET_INVITED_USERS_STATUS } from '../actions/user'
+  SET_CURRENT_ROOM,//Same
+  CHOOSE_CONVERSATION_ROOM,//Same
```
Remove one of the sames

bubuzavrik commented on client/src/components/Messages/Thread/index.js:
```diff
@@ -132,7 +113,22 @@ export const Thread = () => {
       </form>
       <DropDownMenu anchorEl={settingsAnchor} onClose={handleSettingsClose}>
-        {settingsMenu}
+        <MenuItem onClick={handleSettings} key="notes">
```
Use map

emalsidog replied:
Do you mean to create a 4 element array and for each key return a MenuItem? Or you mean just map a settingsMenu?

emalsidog commented:
Fixed your remarks

bubuzavrik commented:
![image](https://user-images.githubusercontent.com/22328251/129535945-0f201340-7c04-4d3d-b5f3-5191aacff244.png)

In the assigned column should be names instead of ids

emalsidog commented:
> ![image](https://user-images.githubusercontent.com/22328251/129535945-0f201340-7c04-4d3d-b5f3-5191aacff244.png)
> 
> In the assigned column should be names instead of ids

Should i use a request to get user by id? For each row?

bubuzavrik commented:
> > ![image](https://user-images.githubusercontent.com/22328251/129535945-0f201340-7c04-4d3d-b5f3-5191aacff244.png)
> > In the assigned column should be names instead of ids
> 
> Should i use a request to get user by id? For each row?

You already have this info. Conversation can be assigned only to members, and you already have members array

emalsidog commented:
Fixed issue with the table but i am not sure, that map is a good idea because I have put different icons and i can not just create array with 4 element and map the same MenuItem

bubuzavrik commented on client/src/components/Messages/Thread/index.js:
```diff
@@ -62,6 +62,25 @@ export const Thread = () => {
+      Leave room
+    </MenuItem>,
+  ];
+
```
You can create array of {key:'', icon:()=>Icon, name:'' }, and map it

emalsidog commented:
Hope its ok :)
